src/mumble/CMakeLists.txt

```diff
@@ -438,13 +439,24 @@ else()
 	target_sources(mumble PRIVATE "SharedMemory_unix.cpp")
 
 	if(NOT APPLE)
-		find_pkg(X11 COMPONENTS Xext REQUIRED)
+		if(x11)
+			find_pkg(X11 COMPONENTS Xext REQUIRED)
+			target_link_libraries(mumble PRIVATE X11::Xext)
+
+			if(xinput2)
+				find_pkg(X11 COMPONENTS Xi REQUIRED)
+				target_link_libraries(mumble PRIVATE X11::Xi)
+			else()
+				target_compile_definitions(mumble PRIVATE "NO_XINPUT2")
+			endif()
 
-		if(xinput2)
-			find_pkg(X11 COMPONENTS Xi REQUIRED)
-			target_link_libraries(mumble PRIVATE X11::Xi)
+			target_sources(mumble
+				PRIVATE
+					"GlobalShortcut_x11.cpp"
+					"GlobalShortcut_x11.h"
+			)
 		else()
-			target_compile_definitions(mumble PRIVATE "NO_XINPUT2")
+			target_compile_definitions(mumble PRIVATE "NO_X11" "NO_XINPUT2")
 		endif()
```

Comment:
Given that the `xinput2` option will be ignored if `x11=OFF`, we should check for that here and error if `xinput2=ON` at this point. At least if the option was explicitly set by the user instead of simply being initialized to its default value.